### PR TITLE
First draft GA blog

### PR DESCRIPTION
**Pull request by ryan-storey**

### CODE REVIEW

File: posts/2022-03-15-ga-release-post.adoc [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, since 22.0.0.2 we have published a brand new Open Liberty guide on containerizing microservices with Podman!
+
+In link:{url-about}[Open Liberty] 22.0.0.3:
+
+* <<sql, Allow SQL Operations to be retried in transaction recovery logs>>
+
+Along with the new features and functions added to the runtime, we’ve also made <<guides, updates to our guides>>.
+
+View the list of fixed bugs in link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22.0.0.3_NO_PERIODS+label%3A%22release+bug%22[22.0.0.3].
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]: Fix link

File: posts/2022-03-15-ga-release-post.adoc [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to the Knowledge Center article link:https://www.ibm.com/support/pages/websphere-application-server-storing-liberty-profile-transaction-logs-relational-database[Storing Liberty profile transaction logs in a relational database].
+
+Add the feature to the `server.xml`:
+
+[#bugs]
+== Notable bugs fixed in this release
+
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22.0.0.3_NO_PERIODS+label%3A%22release+bug%22[full list of bugs fixed in 22.0.0.3].
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]: Fix link

File: posts/2022-03-15-ga-release-post.adoc [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+recoverOnStartup="true"
+waitForRecovery="false"
+heuristicRetryInterval="10"
+enableLogRetries="true"
+/>
+----
+
+By default the new functionality is disabled.
+
+For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to the Knowledge Center article link:https://www.ibm.com/support/pages/websphere-application-server-storing-liberty-profile-transaction-logs-relational-database[Storing Liberty profile transaction logs in a relational database].
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]:
```suggestion
  For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:https://openliberty.io/docs/22.0.0.2/transaction-service.html#log[Transaction log management] doc.
```

File: posts/2022-03-15-ga-release-post.adoc [hunk 4]
```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: "TITLE"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Retrying SQL operations in transaction recovery logs and more in Open Liberty 22.0.0.3 - OpenLiberty.io
+seo-description: Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, since 22.0.0.2 we have published a brand new Open Liberty guide on containerizing microservices with Podman!
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]:
```suggestion
seo-description: Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, we have published a brand new Open Liberty guide on containerizing microservices with Podman!
```

File: posts/2022-03-15-ga-release-post.adoc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+blog_description: Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, since 22.0.0.2 we have published a brand new Open Liberty guide on containerizing microservices with Podman!
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]:
```suggestion
blog_description: Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, we have published a brand new Open Liberty guide on containerizing microservices with Podman!
```

File: posts/2022-03-15-ga-release-post.adoc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Retrying SQL operations in transaction recovery logs and more in Open Liberty 22.0.0.3
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, since 22.0.0.2 we have published a brand new Open Liberty guide on containerizing microservices with Podman!
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]:
```suggestion
Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, we have published a brand new Open Liberty guide on containerizing microservices with Podman!
```

**Review by mbroz2 — CHANGES_REQUESTED** [review]

File: posts/2022-03-15-ga-release-post.adoc [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+recoverOnStartup="true"
+waitForRecovery="false"
+heuristicRetryInterval="10"
+enableLogRetries="true"
+/>
+----
+
+By default the new functionality is disabled.
+
+For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:https://openliberty.io/docs/22.0.0.2/transaction-service.html#log[Transaction log management] doc.
```
Comment (dmuelle) [inline posts/2022-03-15-ga-release-post.adoc]: use `latest` in the docs link unless you are targeting a specific version:

link:https://openliberty.io/docs/latest/transaction-service.html#log[Transaction log management]
or use relative URL
link:{url-prefix}/docs/latest/transaction-service.html#log[Transaction log management]

File: posts/2022-03-15-ga-release-post.adoc [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+[#sql]
+=== Allow SQL Operations to be retried in transaction recovery logs
+
+In Liberty, transaction services are implicitly activated when you specify features that require the use of transactions; for example, `jpa-2.0`, `jdbc-4.0`, and `wasJmsServer-1.0`.
```
Comment (dmuelle) [inline posts/2022-03-15-ga-release-post.adoc]: suggested edit:

In Liberty, transaction services are implicitly activated when you specify features that use transactions, such as  `jpa-2.0`, `jdbc-4.0`, and `wasJmsServer-1.0`.

File: posts/2022-03-15-ga-release-post.adoc [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+When Liberty detects a failure in writing to its transaction recovery logs, it will invalidate the logs and disallow further global transactions. In order to do further transactional work, the customer must restart the server. This means applications have to be reloaded when the server restarts. This can take a considerable amount of time for some applications.
```
Comment (dmuelle) [inline posts/2022-03-15-ga-release-post.adoc]: suggested edit:

When Liberty detects a failure to write to its transaction recovery logs, it invalidates the logs and disallows further global transactions. To resume transactional work, you have to restart the server and reload any applications, which can take considerable time in some cases.

File: posts/2022-03-15-ga-release-post.adoc [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+Under certain circumstances, such as when the recovery log failure is caused by transient database connection errors, the condition can be handled by retrying the database operation thus avoiding the need to shutdown the server and reload the apps.
+
+A new transaction service configuration attribute has been introduced to the transaction stanza in the Liberty `server.xml` file that allows these transaction log SQL operations to be retried. It is named `enableLogRetries` and should be set to `true` to enable retries. For example,
```
Comment (dmuelle) [inline posts/2022-03-15-ga-release-post.adoc]: A new  attribute is available for the `transaction` configuration element in the Liberty `server.xml` file that enables these transaction log SQL operations to be retried.

File: posts/2022-03-15-ga-release-post.adoc [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+heuristicRetryInterval="10"
+enableLogRetries="true"
+/>
+----
+
+By default the new functionality is disabled.
+
+For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:https://openliberty.io/docs/22.0.0.2/transaction-service.html#log[Transaction log management] doc.
+
+Add the feature to the `server.xml`:
```
Comment (dmuelle) [inline posts/2022-03-15-ga-release-post.adoc]: should this be here? Looks like somethings missing

File: posts/2022-03-15-ga-release-post.adoc [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
++
+In 22.0.0.2, `JWT` access token inbound propagation failed when a `JWT` sent as segments starts with `Bearer`. Liberty has support to accept multiple headers from `WebSeal` that include portions of a `JWT` access token. The first header, `Authorization-segments`, for example, indicates how many "n" segments follow. The rest of the headers `Authorization-1` through `Authorization-n` contain the `JWT` access token. When `Authorization-1` started with `Bearer`, the access token could not be parsed correctly for inbound propagation. If JWT segments are sent and the first header, `Authorization-segments-1`, starts with `Bearer`, it should be processed correctly as when a JWT is sent using the `Authorization: Bearer`.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19960[OpenID Connect: Double URL Encoded State Parameter in Redirect location]
++
+Previously, when receiving a state parameter from the RP it appears Liberty returned the state double encoded. The issue appears to occur when processing the liberty page to authorize the RP:
```
Comment (dmuelle) [inline posts/2022-03-15-ga-release-post.adoc]: spell out "relying party (RP)" on first mention

**Review by dmuelle — COMMENTED** [review]
ID review- made a few suggestions, lmk if you have any questions. Thanks

File: posts/2022-03-15-ga-release-post.adoc [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+heuristicRetryInterval="10"
+enableLogRetries="true"
+/>
+----
+
+By default the new functionality is disabled.
+
+For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:{url-prefix}/docs/latest/transaction-service.html#log[Transaction log management] doc.
+
+Add the feature to the `server.xml`:
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]:
```suggestion
```

File: posts/2022-03-15-ga-release-post.adoc [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+[#sql]
+=== Allow SQL Operations to be retried in transaction recovery logs
+
+In Liberty, transaction services are implicitly activated when you specify features that use transactions, such as  `jpa-2.0`, `jdbc-4.0`, and `wasJmsServer-1.0`.
+
+When Liberty detects a failure in writing to its transaction recovery logs, it will invalidate the logs and disallow further global transactions. In order to do further transactional work, the customer must restart the server. This means applications have to be reloaded when the server restarts. This can take a considerable amount of time for some applications.
```
Comment (mbroz2) [inline posts/2022-03-15-ga-release-post.adoc]:
```suggestion
When Liberty detects a failure to write to its transaction recovery logs, it invalidates the logs and disallows further global transactions. To resume transactional work, you have to restart the server and reload any applications, which can take considerable time in some cases.
```

**Review by mbroz2 — APPROVED** [review]

**Pull request merged**